v0/destinations/kustomer/transform.js

```diff
@@ -0,0 +1,233 @@
+        userExists = true;
+        targetUrl = `${BASE_ENDPOINT}/v1/customers/${response.data.data.id}?replace=false`;
+      }
+    }
+    // If userExists flag is false
+    // If Kustomer has userId we search using userId as externalId if user
+    // is present or not. If yes then we mark it for update.
+    if (!userExists && getFieldValueFromMessage(message, "userIdOnly")) {
+      response = await fetchKustomer(
+        `${BASE_ENDPOINT}/v1/customers/externalId=${getFieldValueFromMessage(
+          message,
+          "userIdOnly"
+        )}`,
+        destination
+      );
+      // If Kustomer Exists url to user : Update kustomer
+      if (response && response.status === 200 && response.data.data.id) {
+        userExists = true;
+        targetUrl = `${BASE_ENDPOINT}/v1/customers/${response.data.data.id}?replace=false`;
+      }
+    }
+    // If userExists flag is still false
+    // and the Kustomer has anonymousId we search using anonymousId as externalId.
+    // If present we mark it for update
+    if (!userExists && get(message, "anonymousId")) {
+      response = await fetchKustomer(
+        `${BASE_ENDPOINT}/v1/customers/externalId=${get(
+          message,
+          "anonymousId"
+        )}`,
+        destination
+      );
+      // If Kustomer Exists url to user : Update kustomer
+      if (response && response.status === 200 && response.data.data.id) {
+        userExists = true;
+        targetUrl = `${BASE_ENDPOINT}/v1/customers/${response.data.data.id}?replace=false`;
+      }
+    }
+    // URL to use for creating new Kustomer
+    if (!userExists) {
+      targetUrl = `${BASE_ENDPOINT}/v1/customers`;
+    }
+    payload = constructKustomerPayload(message, category);
+  }
+  // Section responsible for handling screen, page, and track type of
+  // events. An userId, or anonymous id of the user who is already
+  // identified is required for logging events into their timeline.
+  // -----------------------------------------------------------
+  // Ref: https://apidocs.kustomer.com/#fe1b29a6-7f3c-40a7-8f54-973ecd0335e8
+  // Ref: https://apidocs.kustomer.com/#0b0da19f-fca2-401d-af78-5d054c75a9b2
+  else {
+    targetUrl = `${BASE_ENDPOINT}/v1/tracking/identityEvent`;
+    const eventPayload = constructPayload(
+      message,
+      MAPPING_CONFIG[category.name]
+    );
+    // eslint-disable-next-line default-case
+    switch (message.type.toLowerCase()) {
+      case EventType.PAGE:
+        if (destination.Config.genericPage) {
+          eventPayload.name = "Web-Page-Viewed";
+        }
+        break;
+      case EventType.SCREEN:
+        if (destination.Config.genericScreen) {
+          eventPayload.name = "Screen-Viewed";
+        }
+        break;
+    }
+    removeUndefinedAndNullValues(eventPayload);
+    validateEvent(eventPayload);
+    payload = {
+      identity: {
+        externalId: getFieldValueFromMessage(message, "userId")
```

Comment:
Do we have any challenges here for the old users?

Reply:
we are expecting to track with userid as externalid -- so externalid which is used to map user to dest is required as userid for tracking events